builtin/logical/pkiext/pkiext_binary/acme_test.go

```diff
@@ -65,6 +81,161 @@ func Test_ACME(t *testing.T) {
+type caddyConfig struct {
+	Hostname  string
+	Directory string
+	CACert    string
+	EABID     string
+	EABKey    string
+}
+
+// SubtestACMECaddy returns an ACME test for Caddy using the provided template.
+func SubtestACMECaddy(configTemplate string, enableEAB bool) func(*testing.T, *VaultPkiCluster) {
+	return func(t *testing.T, cluster *VaultPkiCluster) {
+		ctx := context.Background()
+
+		// Roll a random run ID for mount and hostname uniqueness.
+		runID, err := uuid.GenerateUUID()
+		require.NoError(t, err, "failed to generate a unique ID for test run")
+		runID = strings.Split(runID, "-")[0]
+
+		// Create the PKI mount with ACME enabled
+		pki, err := cluster.CreateAcmeMount(runID)
+		require.NoError(t, err, "failed to set up ACME mount")
+
+		// Conditionally enable EAB and retrieve the key.
+		var eabID, eabKey string
+		if enableEAB {
+			err = pki.UpdateAcmeConfig(true, map[string]interface{}{
+				"eab_policy": "new-account-required",
+			})
+			require.NoError(t, err, "failed to configure EAB policy in PKI mount")
+
+			eabID, eabKey, err = pki.GetEabKey("acme/")
+			require.NoError(t, err, "failed to retrieve EAB key from PKI mount")
+		}
+
+		directory := fmt.Sprintf("https://%s:8200/v1/%s/acme/directory", pki.GetActiveContainerIP(), runID)
+		vaultNetwork := pki.GetContainerNetworkName()
+		t.Logf("dir: %s", directory)
+
+		logConsumer, logStdout, logStderr := getDockerLog(t)
+
+		sleepTimer := "45"
+
+		// Kick off Caddy container.
+		t.Logf("creating on network: %v", vaultNetwork)
+		caddyRunner, err := hDocker.NewServiceRunner(hDocker.RunOptions{
+			// TODO: Replace with pull-through cache. - schultz
+			ImageRepo:     "library/caddy",
+			ImageTag:      "latest",
+			ContainerName: fmt.Sprintf("caddy_test_%s", runID),
+			NetworkName:   vaultNetwork,
+			Ports:         []string{"80/tcp", "443/tcp", "443/udp"},
+			Entrypoint:    []string{"sleep", sleepTimer},
+			LogConsumer:   logConsumer,
+			LogStdout:     logStdout,
+			LogStderr:     logStderr,
+		})
+		require.NoError(t, err, "failed creating caddy service runner")
+
+		caddyResult, err := caddyRunner.Start(ctx, true, false)
+		require.NoError(t, err, "could not start Caddy container")
+		require.NotNil(t, caddyResult, "could not start Caddy container")
+
+		defer caddyRunner.Stop(ctx, caddyResult.Container.ID)
+
+		networks, err := caddyRunner.GetNetworkAndAddresses(caddyResult.Container.ID)
+		require.NoError(t, err, "could not read caddy container's IP address")
+		require.Contains(t, networks, vaultNetwork, "expected to contain vault network")
+
+		ipAddr := networks[vaultNetwork]
+		hostname := fmt.Sprintf("%s.dadgarcorp.com", runID)
+
+		err = pki.AddHostname(hostname, ipAddr)
+		require.NoError(t, err, "failed to update vault host files")
+
+		// Copy the Vault cluster's CA certificate over to the Caddy container.
+		caCertCtx := hDocker.NewBuildContext()
+		caCertCtx["vault_ca_cert.crt"] = hDocker.PathContentsFromString(string(cluster.GetCACertPEM()))
```

Comment:
This and the push below could be combined fwiw I believe.

Reply:
Oh, good catch! I'll combine them